enterprise/cmd/lsif-index-tester/main.go

```diff
@@ -0,0 +1,363 @@
+type UsageStats struct {
+	// Memory usage in kilobytes by child process.
+	memory int64
+}
+
+type PassedTest struct {
+	Name string
+}
+
+type FailedTest struct {
+	Name string
+	Diff string
+}
+
+type TestFileResult struct {
+	Name   string
+	Passed []PassedTest
+	Failed []FailedTest
+}
+
+type TestSuiteResult struct {
+	FileResults []TestFileResult
+}
+
+var directory string
+var raw_indexer string
+var debug bool
+
+// TODO: Do more monitoring of the process.
+// var monitor bool
+
+func logFatal(msg string, args ...interface{}) {
+	log15.Error(msg, args...)
+	os.Exit(1)
+}
+
+func main() {
+	flag.StringVar(&directory, "dir", ".", "The directory to run the test harness over")
+	flag.StringVar(&raw_indexer, "indexer", "", "The name of the indexer that you want to test")
+	flag.BoolVar(&debug, "debug", false, "Enable debugging")
+
+	flag.Parse()
+
+	if debug {
+		log15.Root().SetHandler(log15.LvlFilterHandler(log15.LvlDebug, log15.StdoutHandler))
+	} else {
+		log15.Root().SetHandler(log15.LvlFilterHandler(log15.LvlError, log15.StdoutHandler))
+	}
+
+	if raw_indexer == "" {
+		logFatal("Indexer is required. Pass with --indexer")
+	}
+
+	var indexer = strings.Split(raw_indexer, " ")
+
+	log15.Info("Starting Execution: ", "base directory", directory, "indexer", raw_indexer)
+
+	testContext := context.Background()
+	err := testDirectory(testContext, indexer, directory)
+	if err != nil {
+		logFatal("Failed with", "err", err)
+	}
+}
+
+func testDirectory(ctx context.Context, indexer []string, directory string) error {
+	files, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return err
+	}
+
+	type ChannelResult struct {
+		result ProjectResult
+		err    error
+	}
+
+	resultChan := make(chan ChannelResult, len(files))
+	var wg sync.WaitGroup
+
+	for _, f := range files {
+		wg.Add(1)
+
+		go func(name string) {
+			defer wg.Done()
+
+			projResult, err := testProject(ctx, indexer, path.Join(directory, name), name)
+
+			resultChan <- ChannelResult{
+				result: projResult,
+				err:    err,
+			}
+		}(f.Name())
+
+	}
+
+	wg.Wait()
+	close(resultChan)
+
+	allPassed := true
+	for chanResult := range resultChan {
+		if chanResult.err != nil {
+			log15.Warn("Failed to run test. Got err:", "error", chanResult.err)
+			continue
+		}
+
+		for _, fileResult := range chanResult.result.testResult.FileResults {
+			if len(fileResult.Failed) > 0 {
+				allPassed = false
+				for _, failed := range fileResult.Failed {
+					fmt.Printf("Failed test: File: %s, Name: %s\nDiff: %s\n", fileResult.Name, failed.Name, failed.Diff)
+				}
+			}
+		}
+	}
+
+	if !allPassed {
+		return errors.New("Failed some tests. Try again later :)")
+	}
+
+	return nil
+}
+
+func testProject(ctx context.Context, indexer []string, project, name string) (ProjectResult, error) {
+	output, err := setupProject(project)
+	if err != nil {
+		return ProjectResult{name: name, success: false, output: string(output)}, err
+	} else {
+		log15.Debug("... Generated compile_commands.json")
+	}
```

Comment:
```suggestion
	log15.Debug("... Generated compile_commands.json")
```

It's a lint error to have an else clause to an if that does an unconditional return (this may have been flagged in CI).